frontend/src/pages/Projects/Project/ProjectDetails/BudgetSection/FormConfig.ts

```diff
@@ -2,14 +2,125 @@ import { AxiosResponse } from "axios";
 import { UseQueryResult } from "@tanstack/react-query";
 import { IEditField } from "types";
 import { useParams } from "react-router-dom";
-import { object, string } from "yup";
+import { object, string, number } from "yup";
+import { FormikValues } from "formik";
+import _ from "lodash";
 
-const validationSchema = object({
-  stob: string()
-    .min(4, "Must contain exactly 4 alphanumeric characters.")
-    .max(4, "Must contain exactly 4 alphanumeric characters.")
-    .matches(/^[A-Za-z0-9]{4}$/, "Must contain exactly 4 alphanumeric characters."),
-});
+interface IRecoveredQuarterAmounts {
+  q1_amount: string;
+  q2_amount: string;
+  q3_amount: string;
+  q4_amount: string;
+  [key: string]: string;
+}
+
+/**
+ * Updates the values in the `recoveredQuarterAmounts` object with the values from the `newValue` object.
```

Comment:
this seems to be the exact same description for the `updateQuarterAmounts` function